File: Portfolio/Portfolio.Blazor/Data/Entities/SnippetFile.cs
namespace Portfolio.Blazor.Data.Entities;

public sealed class SnippetFile
{
    public int Id { get; set; }
    public int SnippetId { get; set; }
    public string Path { get; set; } = string.Empty;
    public string? Language { get; set; }
    public string Content { get; set; } = string.Empty;
    public int Order { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Snippet? Snippet { get; set; }
}

File: laravel/app/Models/SnippetFile.php
<?php

namespace App\Models;

use App\Models\Concerns\Auditable;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class SnippetFile extends Model
{
    use Auditable, HasFactory;

    protected $fillable = ['snippet_id', 'path', 'language', 'content', 'order'];

    /**
     * @return BelongsTo<Snippet, $this>
     */
    public function snippet(): BelongsTo
    {
        return $this->belongsTo(Snippet::class);
    }
}
